hazelcast/src/main/java/com/hazelcast/kubernetes/KubernetesClient.java

```diff
@@ -619,4 +693,116 @@ public String toString() {
+            String watchType = jsonObject.getString("type", null);
+            RuntimeContext ctx = null;
+            switch (watchType) {
+                case "MODIFIED":
+                    ctx = extractSts(sts);
+                    latestResourceVersion = ctx.getResourceVersion();
+                    break;
+                case "DELETED":
+                    ctx = extractSts(sts);
+                    latestResourceVersion = ctx.getResourceVersion();
+                    ctx = new RuntimeContext(0, ctx.getReadyReplicas(),
+                            ctx.getCurrentReplicas(), ctx.getResourceVersion());
+                    break;
+                case "ADDED":
+                    throw new IllegalStateException("A new sts with same name as this cannot be added");
+                default:
+                    LOGGER.info("Unknown watch type " + watchType + ", complete message:\n" + message);
+            }
+            if (latestRuntimeContext != null && ctx != null) {
+                LOGGER.info("Updating cluster topology tracker with previous: "
+                    + latestRuntimeContext + ", updated: " + ctx);
+                // Check for stable cluster spec size and missing member rejoined while another member is preparing
+                // for restart coalesced event. This typically occurs during a rolling restart (kubectl rollout restart):
+                // member A (previously restarted) is rejoining the cluster, cluster is again complete but member B is
+                // signalled to terminate at the same time.
+                // The form of the watch event from Kubernetes API is like this:
+                // StatefulSet.spec (previous) == StatefulSet.spec (current) -- no change to requested cluster size
+                // StatefulSetStatus.readyReplicas (previous) < StatefulSetStatus.readyReplicas (current) -- member rejoined
+                // StatefulSet.spec (current) == StatefulSetStatus.readyReplicas (current) -- cluster reached requested size
+                // StatefulSetStatus.replicas (previous) > StatefulSetStatus.replicas (current) -- next pod is getting terminated
+                if (!deliverCoalescedEvents
+                    && ctx.getSpecifiedReplicaCount() == latestRuntimeContext.getSpecifiedReplicaCount()
+                    && ctx.getReadyReplicas() > latestRuntimeContext.getReadyReplicas()
+                    && ctx.getReadyReplicas() == ctx.getSpecifiedReplicaCount()
+                    && ctx.getCurrentReplicas() < latestRuntimeContext.getCurrentReplicas()) {
+                    // unbundle this into two events
+                    // 1. ready replicas == spec size with same "currentReplicas" as previous
+                    clusterTopologyIntentTracker.update(latestRuntimeContext.getSpecifiedReplicaCount(),
+                            ctx.getSpecifiedReplicaCount(), ctx.getReadyReplicas(), latestRuntimeContext.getCurrentReplicas());
+                    Thread.yield();
```

Comment:
Why we need `yield` here?

Reply:
I added it as a means to give a chance to actions submitted to `clusterTopologyExecutor` to execute (especially in highly loaded instances), but I don't see it as necessary and will remove it in a next commit.

Reply:
removed in 3d3f4701